src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -0,0 +1,193 @@
+                _userNotificationServices.ShowWarning(failureMessage);
+                return;
+            }
+
+            // Rename the file
+            await CPSRenameAsync(context, node, value);
+
+            //Check if there are any symbols that need to be renamed
+            if (documentRenameResult.ApplicableActions.IsEmpty)
+                return;
+
+            // Ask if the user wants to rename the symbol
+            bool userWantsToRenameSymbol = await CheckUserConfirmation(oldName);
+            if (!userWantsToRenameSymbol)
+                return;
+
+            _threadingService.RunAndForget(async () =>
+            {
+                // TODO - implement PublishAsync() to sync with LanguageService
+                // https://github.com/dotnet/project-system/issues/3425)
+                // await _languageService.PublishAsync(treeVersion);
+                var stageStatus = (await _operationProgressService.GetValueAsync()).GetStageStatus(CommonOperationProgressStageIds.Intellisense);
+                await stageStatus.WaitForCompletionAsync();
+
+                // Apply actions and notify other VS features
+                CodeAnalysis.Solution? currentSolution = GetCurrentProject()?.Solution;
+                if (currentSolution == null)
+                {
+                    return;
+                }
+                string renameOperationName = string.Format(CultureInfo.CurrentCulture, VSResources.Renaming_Type_from_0_to_1, oldName, value);
+                (WaitIndicatorResult result, CodeAnalysis.Solution renamedSolution) = _waitService.WaitForAsyncFunctionWithResult(
+                                title: VSResources.Renaming_Type,
+                                message: renameOperationName,
+                                allowCancel: true,
+                                token => documentRenameResult.UpdateSolutionAsync(currentSolution, token));
+
+                // Do not warn the user if the rename was cancelled by the user	
+                if (result.WasCanceled())
+                {
+                    return;
+                }
+
+                await _projectVsServices.ThreadingService.SwitchToUIThread();
+                if (!_roslynServices.ApplyChangesToSolution(currentSolution.Workspace, renamedSolution))
+                {
+                    string failureMessage = string.Format(CultureInfo.CurrentCulture, VSResources.RenameSymbolFailed, oldName);
+                    _userNotificationServices.ShowWarning(failureMessage);
+                }
+                return;
+            }, _unconfiguredProject);
+
+        }
+
+        protected virtual string GetFailureMessageCannotApply(CodeAnalysis.Rename.Renamer.RenameDocumentActionSet? documentRenameResult)
+        {
+            return "Cannot apply " +
+                documentRenameResult?.ApplicableActions.First().GetDescription(CultureInfo.CurrentCulture);
```

Comment:
"CurrentUICulture" represents the localized language.
"CurrentCulture" represents the formatting/parsing language.

Reply:
done. Changed to use localized language